compiler/rustc_privacy/src/lib.rs

```diff
@@ -1358,7 +1361,20 @@ impl<'tcx> Visitor<'tcx> for TypePrivacyVisitor<'tcx> {
     fn visit_item(&mut self, item: &'tcx hir::Item<'tcx>) {
         let orig_current_item = mem::replace(&mut self.current_item, item.owner_id.def_id);
         let old_maybe_typeck_results = self.maybe_typeck_results.take();
-        intravisit::walk_item(self, item);
+        rustc_ty_utils::sig_types::walk_types(self.tcx(), item.owner_id.def_id, self);
+        match item.kind {
+            hir::ItemKind::Macro(_, _) => intravisit::walk_item(self, item),
+            hir::ItemKind::Static(_, _, body_id)
+            | hir::ItemKind::Const(_, _, body_id)
+            | hir::ItemKind::Fn(_, _, body_id) => self.visit_nested_body(body_id),
+            hir::ItemKind::Impl(hir::Impl { of_trait: Some(tr), .. }) => {
+                self.span = tr.path.span;
+                let trait_ref =
+                    self.tcx().impl_trait_ref(item.owner_id.def_id).unwrap().instantiate_identity();
+                self.visit_def_id(trait_ref.def_id, "trait", &trait_ref.print_only_trait_path());
+            }
+            _ => {}
+        }
```

Comment:
Wait, if `fn visit_ty` with `hir_ty_to_ty` still exists, then why this change is needed?
All the signature types will be visited by `visit_ty` anyway.

Reply:
Oh yea that was all just dead code. I removed it